drivers/staging/vc04_services/bcm2835-codec/bcm2835-v4l2-codec.c

```diff
@@ -3332,11 +3415,12 @@ static int bcm2835_codec_open(struct file *file)
 	break;
 	case DECODE:
 	{
-		v4l2_ctrl_handler_init(hdl, 1);
+		v4l2_ctrl_handler_init(hdl, 1 + dev->supported_fmts[0].num_entries * 2);
```

Comment:
This is a little annoying as there are formats that won't add the associated controls, but such is life other than a loop to return the number of controls, and a second loop that then creates the controls.

Reply:
I thought about that but decided that there was nowhere near enough gain. The added code size would probably be bigger than the wasted memory.